src/python/pants/core/goals/test.py

```diff
@@ -87,6 +87,17 @@ def from_fallible_process_result(
             xml_results=xml_results,
         )
 
+
+@dataclass(frozen=True)
+class EnrichedTestResult(EngineAware):
+    exit_code: Optional[int]
```

Comment:
Stu pointed out in https://github.com/pantsbuild/pants/pull/10632#issuecomment-675033366 that using the null pattern with `Optional[int]` to represent "skipped" is not great modeling. Alternatively, we could have an `OptionalTestResult` top-level type to keep the model more correct.

I think it would indeed be more correct, but my concern is that it makes `test.py` and each test implementation a lot clunkier.